### PR TITLE
Check for md5 checksums if remote uses that

### LINKED ISSUE #245: overwrite does wrong checksum validation
**Describe the bug**
When setting the overwrite it will do a checksum test but it does the sha2 checksum and not the md5 that is being used in the iRODS instance I have access to.

**To Reproduce**
Make sure your irods uses the md5 checksum
Download a file with overwrite...

```python
        if not (
            local_path.is_file() and (calc_checksum(irods_path) == calc_checksum(local_path))
        ):

# Will call

    if isinstance(filepath, IrodsPath):
        return filepath.checksum
    f_hash=sha256()
    memv=memoryview(bytearray(128*1024))
    with open(filepath, 'rb', buffering=0) as file:
        for item in iter(lambda : file.readinto(memv), 0):
            f_hash.update(memv[:item])
    return f"sha2:{str(base64.b64encode(f_hash.digest()), encoding='utf-8')}"
```

**Provide some information:**
 - OS: Mac osx
 - Software version or branch you are working on
 - python version 3.12.4
 - iRODS version 
 - python-irodsclient version (`python -c "import irods; print(irods.__version__)"` )

### PR DESCRIPTION
**Pull request by qubixes**
Fixes #245

### CODE REVIEW

File: ibridges/util.py [hunk 1]
```diff
@@ -160,9 +163,34 @@ def calc_checksum(filepath: Union[Path, str, IrodsPath]):
...
+    if checksum_type == "sha2":
+        f_hash = sha256()
+    else:
+        f_hash = md5()
     memv=memoryview(bytearray(128*1024))
     with open(filepath, 'rb', buffering=0) as file:
         for item in iter(lambda : file.readinto(memv), 0):
             f_hash.update(memv[:item])
+    if checksum_type == "md5":
+        return f"md5:{f_hash.hexdigest()}"
```
Comment (chStaiger) [inline ibridges/util.py:175]: The native iRODS checksum in the iCAT not use the prefix `md5:` but just returns the md5sum. So will this work?

File: ibridges/util.py [hunk 2]
```diff
@@ -160,9 +163,34 @@ def calc_checksum(filepath: Union[Path, str, IrodsPath]):
...
+        Local path to compute the checksum for.
+
+    Returns
+    -------
+        Whether the two have equal checksums. The type of checksum done
+        depends on what is configured on the remote server.
+
+    """
+    remote_check = calc_checksum(remote_path)
+    local_check = calc_checksum(local_path, checksum_type=remote_check.split(":")[0])
```
Comment (chStaiger) [inline ibridges/util.py:195]: Ah, now I see. You are introducing the `md5:` prefix to unify the code for the two checksum types.

**Review by chStaiger — COMMENTED** [review]
Should we add unit tests for the checksum calculation, have a new option in the config for the tests `checksum_type`?

File: ibridges/util.py (hunk 1 above)
Comment (qubixes) [reply ibridges/util.py:175]: Well, no then it's not going to work. Can we test md5 checksums on both YoDa and the native iRODS integration test?

**Pull request merged**